crates/iceberg/src/transaction.rs

```diff
@@ -1167,6 +1220,248 @@ impl<'a> ReplaceSortOrderAction<'a> {
     }
 }
 
+/// Transaction action for rewriting files.
+#[allow(dead_code)]
+pub struct RewriteFilesAction<'a> {
```

Comment:
Maybe we should put these added code (it's a lot) to a separate file. 
There are some ongoing refactor upstream: https://github.com/apache/iceberg-rust/issues/298

Reply:
Oh, this file is already split upstream: https://github.com/apache/iceberg-rust/pull/1080

Reply:
sure, Should we port this pr directly to the dev branch? https://github.com/apache/iceberg-rust/pull/1080

Reply:
+1 to port the reactor to avoid future conflicts , or just do another rebase first?

Reply:
Support rebase. Can you evaluate the difficulty of rebase? @xxchan @ZENOTME

Reply:
> Support rebase. Can you evaluate the difficulty of rebase? @xxchan @ZENOTME

I'm concerned rebase will cause the extra effort to fix the conflict. We can port first and merge this PR. In addition, rebase after sync merge append and remove snapshot action with upstream will make rebasing them easier in future.